app/main.py

```diff
@@ -1,3 +1,68 @@
+import math
+
+
 class Vector:
-    # write your code here
-    pass
+    def __init__(
+            self,
+            x: float,
+            y: float
+    ):
+        self.x = round(x, 2)
+        self.y = round(y, 2)
+
+    def __add__(self, other):
+        return Vector(
+            x=self.x + other.x,
+            y=self.y + other.y
+        )
+
+    def __sub__(self, other):
+        return Vector(
+            x=self.x - other.x,
+            y=self.y - other.y
+        )
+
+    def __mul__(self, other):
+        if isinstance(other, Vector):
+            dot_product = (self.x * other.x) + (self.y * other.y)
+            return dot_product
+        return Vector(
+            x=self.x * other,
+            y=self.y * other
+        )
+
+    @staticmethod
```

Comment:
you should have to use classmethod here, according to the condition, and apply magic method here (__sub__)